runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java

```diff
@@ -177,6 +188,29 @@ public void onGameStateChange(GameStateChanged event)
 			updateWarnedSkills(config.warnForRanged(), Skill.RANGED);
 			updateWarnedSkills(config.warnForMagic(), Skill.MAGIC);
 		}
+
+		if (event.getGameState() == GameState.LOGIN_SCREEN ||
+			event.getGameState() == GameState.HOPPING ||
+			event.getGameState() == GameState.CONNECTION_LOST)
+		{
+			loginMessageSent = false;
+			messageTickGuarded = true;
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick e)
+	{
+		// Utilizes GameTicks, seeing as variables are set PRIOR to a GameTick actually being ran.
+		// The messages are guarded by this tick-set, therefore, the messages aren't spammed upon
+		// variable setting and then once all variables are set, this code block runs and sends
+		// the corresponding login message(s).
+		if (messageTickGuarded)
+		{
+			messageTickGuarded = false;
+			updateWarning(false);
```

Comment:
Why is this new `updateWarning()` call added?

Reply:
It's to send the login messages, seeing as all variables are set and it can NOW dictate what to output.